hooks/invocation/invocation.go

```diff
@@ -1,24 +1,59 @@
 package invocation
 
 import (
+	"encoding/json"
+	"github.com/prebid/prebid-server/metrics"
 	"time"
 )
 
-type Context struct {
-	Endpoint string
-	Timeout  time.Duration
-	// todo: think on adding next fields
-	// debugEnabled
-	// moduleContext
+type InvocationContext struct {
+	DebugEnabled      bool
+	RequestTypeMetric metrics.RequestType
+	moduleContexts    map[string]*ModuleContext
+}
+
+func (ctx *InvocationContext) ModuleContextFor(moduleCode string) *ModuleContext {
+	if mc, ok := ctx.moduleContexts[moduleCode]; ok {
+		return mc
+	}
+
+	emptyCtx := ModuleContext{}
+
+	if ctx.moduleContexts == nil {
+		ctx.moduleContexts = map[string]*ModuleContext{
+			moduleCode: &emptyCtx,
+		}
+	} else {
+		ctx.moduleContexts[moduleCode] = &emptyCtx
+	}
+
+	return &emptyCtx
+}
+
+type HookResponse[T any] struct {
+	Result HookResult[T]
+	Err    error
 }
 
 type HookResult[T any] struct {
-	Reject    bool
-	Mutations []Mutation[T]
-	// todo: think on adding next fields
+	ModuleCode    string
+	ExecutionTime time.Duration
+	Reject        bool
+	NbrCode       int
+	Message       string
+	ChangeSet     *ChangeSet[T]
+	Errors        []string
+	Warnings      []string
+	DebugMessages []string
+	//todo: think on adding next fields
 	// analyticTags
-	// errors
-	// warnings
-	// debug
-	// moduleContext - arbitrary data the hook wishes to pass to downstream hooks of the same module
+}
+
+type ModuleContext struct {
+	Ctx           map[string]interface{} // interface as we do not know exactly how the modules will use their inner context
```

Comment:
This is just for sharing data between hooks of the same module right? I was trying to figure out what kind of data could be stored here but then I noticed your deleted comment just above this line.

Reply:
Correct. Each module will have some own context that can be used in different stages and we decided to give them the freedom to determine how this context would be defined